jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -342,13 +345,21 @@ def calculate_photometry(self, dataset=None, aperture=None, background=None,
             # we can use the pre-cached value
             data = self.dataset.selected_dc_item
 
-        if aperture is not None and aperture not in self.aperture.choices:
-            raise ValueError(f"aperture must be one of {self.aperture.choices}")
+        if aperture is not None:
+            if aperture not in self.aperture.choices:
+                raise ValueError(f"aperture must be one of {self.aperture.choices}")
+
         if aperture is not None or dataset is not None:
             reg = self.aperture._get_spatial_region(subset=aperture if aperture is not None else self.aperture.selected,  # noqa
                                                     dataset=dataset if dataset is not None else self.dataset.selected)  # noqa
+            # determine if a valid aperture (since selected_validity only applies to selected entry)
+            _, _, validity = self.aperture._get_mark_coords_and_validate(selected=aperture)

```

Comment:
I cannot tell which is better. But we can also just grab the 3rd result and not bother with magic underscores.
```suggestion
            validity = self.aperture._get_mark_coords_and_validate(selected=aperture)[2]
```

Reply:
I find the underscores to be more readable myself 🤷‍♂️